Public/Src/Cache/ContentStore/Library/Stores/ContentHashAddressableSnapshot.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using BuildXL.Cache.ContentStore.Hashing;
+
+namespace BuildXL.Cache.ContentStore.Stores
+{
+    /// <summary>
+    /// This class is specifically for usage within <see cref="ContentHashAddressableSnapshot{T}"/>. It is a reference type on purpose, in order to avoid issues
+    /// with the maximum object size limit.
+    /// </summary>
+    /// <typeparam name="T">Type tagged with a hash</typeparam>
+    public class PayloadFromDisk<T> : IComparable<PayloadFromDisk<T>>
+    {
+        /// <summary>
+        /// Hash for the <see cref="Payload"/>
+        /// </summary>
+        public readonly ContentHash Hash;

```

Comment:
I think we generally prefer get-only properties over public fields and I would encourage to use them because they're easier to change in the future and the same in terms of coding.